src/main/java/com/fakng/fakngagrgtr/parser/apple/AppleParser.java

```diff
@@ -0,0 +1,131 @@
+    private static final String ELEMENT_TAG_WITH_PAGE_COUNT = "span";
+    private static final String PATH_TO_VACANCIES = "searchResults";
+    private static final String PATH_TO_FIELD_WITH_JSON_BODY = ".//script";
+    private static final JSONParser jsonParser = new JSONParser(JSONParser.MODE_JSON_SIMPLE);
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+
+    public AppleParser(WebClient htmlWebClient,
+                       CompanyRepository companyRepository,
+                       LocationProcessor locationProcessor,
+                       @Value("${url.apple}") String url) {
+        super(htmlWebClient, companyRepository, locationProcessor);
+        this.url = url;
+    }
+
+    @Override
+    protected String getCompanyName() {
+        return "Apple";
+    }
+
+    @PostConstruct
+    public void init() {
+        initBase();
+    }
+
+    @Override
+    protected List<Vacancy> getAllVacancies() throws Exception {
+        HtmlPage htmlPage = getResponseByPage(0);
+        int countPages = getCountPages(htmlPage);
+        List<Vacancy> allVacancies = new ArrayList<>(getVacanciesFromDoc(htmlPage));
+        for (int i = 1; i < countPages; i++) {
+            htmlPage = getResponseByPage(i);
+            allVacancies.addAll(getVacanciesFromDoc(htmlPage));
+        }
+        return allVacancies;
+    }
+
+    private HtmlPage getResponseByPage(int numberOfPage) throws IOException {
+        return downloadPage(String.format(url, numberOfPage));
+    }
+
+    private int getCountPages(HtmlPage htmlPage) {
+        return Integer.parseInt(htmlPage.getElementsByIdAndOrName(ELEMENT_NAME_WITH_PAGE_COUNT).get(0).getElementsByTagName(ELEMENT_TAG_WITH_PAGE_COUNT).get(1).getVisibleText());
+    }
+
+    private List<Vacancy> getVacanciesFromDoc(HtmlPage htmlpage) throws ParseException, JsonProcessingException {
+        String jsonBody = getJsonBody(htmlpage);
+        return getVacanciesFromJsonBody(jsonBody);
+    }
+
+    private String getJsonBody(HtmlPage htmlPage) {
+        String textContent = ((HtmlScript) htmlPage.getByXPath(PATH_TO_FIELD_WITH_JSON_BODY).get(1)).getTextContent();
+        return textContent.substring(textContent.indexOf('{'), textContent.lastIndexOf('}') + 1);
+    }
+
+    private List<Vacancy> getVacanciesFromJsonBody(String json) throws ParseException, JsonProcessingException {
+        JSONObject jsonBody = parse(json);
+        JSONArray searchResults = (JSONArray) jsonBody.get(PATH_TO_VACANCIES);
+        List<Vacancy> vacancies = new ArrayList<>();
+        for (Object vacancyJson : searchResults) {
+            VacancyApple vacancyApple = toVacancyApple(vacancyJson);
+            vacancies.add(toVacancy(vacancyApple));
+        }
+        return vacancies;
+    }
+
+    private JSONObject parse(String json) throws ParseException {
+        return (JSONObject) jsonParser.parse(json);
+    }
+
+    private VacancyApple toVacancyApple(Object vacancyJson) throws JsonProcessingException {
```

Comment:
Please, remove "Apple" word from the method's name

Reply:
Won't this create confusion between? We have VacancyApple and Vacancy

Reply:
That's why I created 2 methods: toVacancy and toVacancyApple. I could call it VacancyDTO, then wouldn't it create confusion with the rest of the parsers?

Reply:
Probably you won't even need this method, if you get rid of JsonParser object
But in general, we rather name those classes with DTO suffix. So better to follow this naming at least and not to tight ourselves to the domain